### PR TITLE
fix: allow users to export an attachment with filename including a special character

### PR DESCRIPTION
**Pull request by ryounasso**
## Why

On cli-kintone v1 for Windows, if we try to download a file with a name that contains characters that cannot be used for the Windows file name, it fails.

However, there are cases that we actually want to download such files and need to fix.

## What

- Added processing to convert prohibited characters to `_` .
- When running on Windows, the conversion process is performed.
- Add test cases for fixes.

## How to test

- Add test cases for fixes.

## Checklist

- [x] Read [CONTRIBUTING.md](https://github.com/kintone/cli-kintone/blob/main/CONTRIBUTING.md)
- [x] Updated documentation if it is required.
- [x] Added tests if it is required.
- [x] Passed `yarn lint` and `yarn test` on the root directory.

### CODE REVIEW

File: src/record/export/utils/operateFilename.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+// https://learn.microsoft.com/en-us/windows/win32/fileio/naming-a-file
```
Comment (mshrtsr) [inline src/record/export/utils/operateFilename.ts:1]: IMO: You can simplify the file name like `utils/file.ts`

File: src/record/export/usecases/get.ts [hunk 2]
```diff
@@ -111,7 +112,9 @@ const fieldProcessor: (
           const localFilePath = path.join(
             attachmentsDir,
             `${fieldSchema.code}-${recordId}`,
-            fileInfo.name
+            process.platform === "win32"
+              ? sanitizeFilename(fileInfo.name)
+              : fileInfo.name
```
Comment (mshrtsr) [inline src/record/export/usecases/get.ts:117]: MUST: You need to change the same process of the `fieldProcessorInSubtable()` too.

File: src/record/export/usecases/__tests__/get/fixtures/can_download_files_including_special_characters.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+import path from "path";
+
+const fileInfo = {
+  contentType: "text/plain",
+  fileKey: "test-file-key",
+  name: ':"t/e\\|s?*t<>.txt',
+  size: "123456",
+};
+
+export const input: KintoneRecordForResponse[] = [
```
Comment (mshrtsr) [inline src/record/export/usecases/__tests__/get/fixtures/can_download_files_including_special_characters.ts:13]: IMO: I recommend you to add Attachment field in Table to your fixture.

**Review by ueokande — APPROVED** [review]

**Review by tasshi-me — CHANGES_REQUESTED** [review]
@ryounasso 
It's almost LGTM but need to be fixed a little.
I left some comment.

**Comment by mshrtsr** [comment]
In addition, Please add a note to our README.md field description section:)
https://github.com/kintone/cli-kintone#attachment

File: src/record/export/utils/operateFilename.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
...
+export const sanitizeFilename = (filename: string): string => {
```
Comment (hung-cybo) [inline src/record/export/utils/operateFilename.ts:2]: @ryounasso 

IMO: The name of the function is a bit generic, we can consider another specific name, for example: `replaceSpecialCharacters`

File: src/record/export/utils/operateFilename.ts (hunk 1 above)
Comment (ryounasso) [reply src/record/export/utils/operateFilename.ts]: Fixed with 31671f3348b0138d9c98d158321636ac2f2f7588

File: src/record/export/usecases/get.ts (hunk 2 above)
Comment (ryounasso) [reply src/record/export/usecases/get.ts]: Fixed with 42bcc409cfa1b68bf2a8b11eb8d31950c7836fde

File: src/record/export/usecases/__tests__/get/fixtures/can_download_files_including_special_characters.ts (hunk 3 above)
Comment (ryounasso) [reply src/record/export/usecases/__tests__/get/fixtures/can_download_files_including_special_characters.ts:13]: Fixed with 232d881fcbf5b7474965ce29d9752b38d859c6a3

File: src/record/export/utils/operateFilename.ts (hunk 4 above)
Comment (ryounasso) [reply src/record/export/utils/operateFilename.ts]: I agree with your opinion.
So, fixed with d25a2e5d207680b506dee918a484c6d5ce03e0f9

**Comment by ryounasso** [comment]
> In addition, Please add a note to our README.md field description section:) https://github.com/kintone/cli-kintone#attachment

Fixed with 7a870d3b82b5bfb99107c78603f884b59ac9cc9d

File: src/record/export/usecases/__tests__/get/fixtures/can_download_files_including_special_characters.ts (hunk 3 above)
Comment (ryounasso) [reply src/record/export/usecases/__tests__/get/fixtures/can_download_files_including_special_characters.ts:13]: Fixed with 2102c7434d0c101b98707844840f3079322add3c

**Review by tasshi-me — APPROVED** [review]
LGTM 👍

**Pull request merged**